flybywiresim/aircraft
# fix: QFE text missing

Fixes #8478 

## Summary of Changes


* Add the optional parameter `classPrefix` to the FCU's `setTextElementActive()` function
  * This is necessary because, unlike the rest of the text elements in the FCU, these ones are identified by class rather than ID, because there are left and right versions
* Use `setElementVisibility()` instead of manually setting the display value

## Screenshots (if necessary)



### Before:

QNH displayed despite QFE being selected:

![image](https://github.com/flybywiresim/aircraft/assets/4079548/f4692c73-956f-44f0-b289-3ae11fada60a)

QFE missing in lights test:

![image](https://github.com/flybywiresim/aircraft/assets/4079548/5f46e7f7-be7c-4854-b617-c1ccf06da1c9)

### After:

QNH on both FCU and PFD:

![Screenshot 2024-02-19 195319](https://github.com/flybywiresim/aircraft/assets/4079548/baddf271-e523-44b4-b0b8-4d4f07401f5e)

QFE on both FCU and PFD:

![Screenshot 2024-02-19 195332](https://github.com/flybywiresim/aircraft/assets/4079548/0c240a2a-d3d6-4499-8af6-4a8d1b9072b4)

STD on both FCU and PFD:

![Screenshot 2024-02-19 195348](https://github.com/flybywiresim/aircraft/assets/4079548/53f8b8c6-1b32-4ac0-b934-816ac2fbf879)

Lights test:

![Screenshot 2024-02-19 195428](https://github.com/flybywiresim/aircraft/assets/4079548/ae7ce8bd-7e30-4012-8890-09703c4e713a)

## References





This is behaviour that existed previously and began malfunctioning, seemingly starting with #8334 

## Additional context




## Testing instructions


1. Load into a flight cold and dark
2. Connect the external power and turn on the batteries
3. Push in the pressure mode selector and observe that it switches to QFE. Push it again and it switches to QNH. Push it once more and it switches to QFE.
4. Pull the pressure mode selector and observe that it switches to STD
5. Push the pressure mode selector and observe that it returns to QFE, which was the previously selected mode
6. Turn the annunciator light switch to TEST and observe `QFE QNH` and `88.88`


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
